skynet/dgpu/network.py

```diff
@@ -13,40 +13,72 @@
 import anyio
 import httpx
 
-from PIL import Image, UnidentifiedImageError
+from PIL import (
+    Image,
+    # UnidentifiedImageError,  # TODO, remove?
+)
 
 from leap.cleos import CLEOS
 from leap.protocol import Asset
-from skynet.constants import DEFAULT_IPFS_DOMAIN, GPU_CONTRACT_ABI
-
-from skynet.ipfs import AsyncIPFSHTTP, get_ipfs_file
-from skynet.dgpu.errors import DGPUComputeError
-
-
-REQUEST_UPDATE_TIME = 3
-
-
-async def failable(fn: partial, ret_fail=None):
+from skynet.constants import (
+    DEFAULT_IPFS_DOMAIN,
+    GPU_CONTRACT_ABI,
+)
+
+from skynet.ipfs import (
+    AsyncIPFSHTTP,
+    get_ipfs_file,
+)
+# TODO, remove?
+# from skynet.dgpu.errors import DGPUComputeError
+
+
+REQUEST_UPDATE_TIME: int = 3
+
+
+# TODO, consider using the `outcome` lib instead?
+# - it's already purpose built for exactly this, boxing (async)
+#  function invocations..
+# |_ https://outcome.readthedocs.io/en/latest/api.html#outcome.capture
+async def failable(
+    fn: partial,
+    ret_fail=None,
+):
     try:
         return await fn()
-
     except (
         OSError,
         json.JSONDecodeError,
         anyio.BrokenResourceError,
         httpx.ReadError,
         leap.errors.TransactionPushError
-    ) as e:
+    ):
         return ret_fail
 
 
+# TODO, again the prefix XD
+# -[ ] better name then `GPUConnector` ??
+# |_ `Compute[Net]IO[Mngr]`
 class SkynetGPUConnector:
+    '''
+    An API for connecting to and conducting various "high level"
+    network-service operations in the skynet.
+
+    - skynet user account creds
+    - hyperion API
+    - IPFs client
+    - CLEOS client
 
+    '''
     def __init__(self, config: dict):
+        # TODO, why these extra instance vars for an (unsynced)
+        # copy of the `config` state?
         self.account = config['account']
         self.permission = config['permission']
         self.key = config['key']
 
+        # TODO, neither of these instance vars are used anywhere in
+        # methods? so why are they set on this type?
```

Comment:
Haha i hate dict access syntax, we should create a `msgspec` struct for config and just store a copy of that.

Reply:
that would be decent bc then it could be msg-passed between daemons (or there sub actors/processes 😉 😉 ) easily when we start getting to dynamic worker scaling Bo

Reply:
Alright, part of the refactor in #48 was moving to a msgspec struct for config, this fixes this issue